File: Projects/Problem9.java
package Projects;

public class Problem9 {

	public static void main(String args[]) {
		
		double a  =0;
		double b = 0;
		double c = 0;
		
		double aSquared = 0;
		double bSquared = 0;
		double cSquared = 0;
		
		
	/*	A Pythagorean triplet is a set of three natural numbers, a < b < c, 
	 * for which, a2 + b2 = c2

	For example, 32 + 42 = 9 + 16 = 25 = 52.

	There exists exactly one Pythagorean triplet for which a + b + c = 1000.
    Find the product abc.

	
		ORIGINAL ATTEMPT: FAILED
		
		for(int i = 1; i<26; i++) {
			for(int k = 0; k<26; i++) {
				aSquared = Math.pow(i,2);
				bSquared = Math.pow(k,2);
				
				System.out.println("works1");
				cSquared = aSquared + bSquared;
				
				if(cSquared == 25)	break;
				System.out.println("works2");
			}
			
			System.out.println("works3");
			
		}
		
		a = Math.sqrt(aSquared);
		b = Math.sqrt(bSquared);
		c = Math.sqrt(cSquared);*/
		
		int s = 1000;
		
		for(int i =1; i <s/3; i++) {
			for(int j = i; j <s/2; j++) {
				c = s - j - i;
				
				if((j*j) + (i*i) == (c*c)) {
					a =j;
					b = i;
					break;
				}
				if( a == j)	break;
			}
		}
		
		System.out.println(a);
		System.out.println(b);
		System.out.println(c);
		
		System.out.println("hello world");
	}
	
}

File: Projects/Problem9.py
#Problem 9- Project Euler
# A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
#
# a^2 + b^2 = c^2
# For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.
#
# There exists exactly one Pythagorean triplet for which a + b + c = 1000.
# Find the product abc.

def triplet(x,y):
    return (x**2 + y**2)

def construct():
    for i in range(500):
        for j in range(i):
            a = i**2 - j**2
            b =2 *i*j
            c= i**2 + j**2

            if(a+b+c ==1000):    return a*b*c

def main():
    print(construct())

main()
